### DISCUSSION TITLE
Failed to find current location without Google Play Services

### DISCUSSION THREAD

**Discussion by RaptaG** [breezy-weather/breezy-weather#1099] [Q&A]
Errors during refresh

! Android: Failed to find current location
  - Current location

**Comment by papjul** [comment]
Pull to refresh from the app, and tap the "Help" button when you see the snackbar error. There will be a few instructions to try to workaround this issue.

**Reply by RaptaG** [reply]
> Pull to refresh from the app, and tap the "Help" button when you see the snackbar error. There will be a few instructions to try to workaround this issue.

Is this planned to be fixed? The app should support the current location and not have to rely on set locations

**Reply by papjul** [reply]
There is nothing to fix, it works. Please follow instructions to know why it doesn't work on your side.

**Reply by RaptaG** [reply]
> There is nothing to fix, it works. Please follow instructions to know why it doesn't work on your side.

- It has all the permissions it needs
- There are no battery/background restrictions
- It won't work with "Current Location" but instead only works with set locations

Could this be a provider issue?

**Reply by papjul** [reply]
1. Are you inside or outside?
2. Do you have Google Play Services? If so, are they enabled (precise location by Google) with wifi and bluetooth scanning enabled?
3. Are you able to get location in other apps? If so, does it work when you go back to Breezy Weather and retry?

**Reply by RaptaG** [reply]
> 1. Are you inside or outside?
> 
>     2. Do you have Google Play Services? If so, are they enabled (precise location by Google) with wifi and bluetooth scanning enabled?
> 
>     3. Are you able to get location in other apps? If so, does it work when you go back to Breezy Weather and retry?

1. Inside
2. No
3. Yes, no it doesn't

**Reply by papjul** [reply]
How do you get location inside without Google Play Services? You need a network location provider as GPS cannot be reached inside

**Reply by RaptaG** [reply]
> How do you get location inside without Google Play Services? You need a network location provider as GPS cannot be reached inside

/e/OS' default, Mozilla Location Services

**Reply by papjul** [reply]
Works fine on my side with or without Google Play Services, inside, with bluetooth and wifi scanning enabled.

Mozilla Location Services are closed, and they will definitely be archived in July. I believe /e/os is using microG which is interfering with the standard behavior of AOSP/Google Play Services. I cannot help on that matter.

**Reply by RaptaG** [reply]
> Works fine on my side with or without Google Play Services, inside, with bluetooth and wifi scanning enabled.
> 
> Mozilla Location Services are closed, and they will definitely be archived in July. I believe /e/os is using microG which is interfering with the standard behavior of AOSP/Google Play Services. I cannot help on that matter.

The thing is, I manually removed microG since I don't use it.

No worries, I'll figure it out myself👍

**Reply by papjul** [reply]
> The thing is, I manually removed microG since I don't use it.

You said you use Mozilla Location Services, how do you do that without microG?

**Reply by RaptaG** [reply]
> > The thing is, I manually removed microG since I don't use it.
> 
> You said you use Mozilla Location Services, how do you do that without microG?

Is microG required to use MSL?

**Reply by papjul** [reply]
If you are talking about Mozilla Location Services, we never call them directly from app, and Android AOSP does not bundle it, and does not allow an end-user to install a custom network provider. Android only allows you to install a mock location provider (which we support). So I can't confirm it comes from microG, but I can tell you this is not something that comes bundled in either Android or Breezy Weather. Anyway, Mozilla Location Services are closing, so...

By the way, we don't call Google Play Services directly (which microG impersonates), but native Android functions. I don't know if microG also handle that case, and if it is a difference with the other apps you use.

Do you confirm wifi and bluetooth scanning is enabled in Android settings?

**Reply by RaptaG** [reply]
> Do you confirm wifi and bluetooth scanning is enabled in Android settings?

Only WiFi scanning
